js/initcontext.go

```diff
@@ -110,7 +119,47 @@ func newBoundInitContext(base *InitContext, ctxPtr *context.Context, rt *goja.Ru
 		programs:          programs,
 		compatibilityMode: base.compatibilityMode,
 		logger:            base.logger,
+
+		shares:     base.shares,
+		sharesLock: base.sharesLock,
+	}
+}
+
+// XShare is a constructor returning a shareable read-only array (general objects coming)
+// indentified by the name and having their contents be whatever the call returns - again only
+// arrays currently supported.
+func (i *InitContext) XShare(name string, call goja.Callable) goja.Value {
+	i.sharesLock.Lock()
+	defer i.sharesLock.Unlock()
+	value, ok := i.shares[name]
+	if !ok { //nolint:nestif
```

Comment:
predominantly because earlier the code at the end was way longer ... but the nolint is no longer necessary actually .. so maybe the linter checking the unneeded nolints ... has a bug :scream: